webpack/assets/javascripts/react_app/redux/middlewares/ajaxRequestAction.js

```diff
@@ -0,0 +1,32 @@
+import API from '../../API';
+
+const get = (payload, url, store, consts) => {
+  store.dispatch({
+    type: consts.REQUEST,
+    payload,
+  });
+  API.get(url, payload.headers || {}, payload.params || {})
```

Comment:
I would suggest making this more dynamic by passing the `API_REQUEST_TYPE` 
as `API[API_REQUEST_TYPE](url, payload...)`

Reply:
This is the next PR :)
I want to make this PR only for the switch